### PR TITLE
Azure blobfuse plugin support in jobmanager

### PR DESCRIPTION
**Pull request by Anbang-Hu**

### CODE REVIEW

File: src/ClusterManager/dist_pod_template.py [hunk 1]
```diff
@@ -152,9 +153,40 @@ def generate_pods(self, job):
...
 
         return k8s_pods, None
+
+    # TODO: Merge with pod_template.py
+    def generate_secrets(self, job):
+        """generate_plugin_secrets must be called after generate_pods"""
+        assert (isinstance(job, Job))
+        params = job.params
+
+        if "plugin" not in params:
```
Comment (leigaoms) [inline src/ClusterManager/dist_pod_template.py]: "plugins" here

Comment (Anbang-Hu) [reply src/ClusterManager/dist_pod_template.py]: fixed

File: src/ClusterManager/pod_template.py [hunk 2]
```diff
@@ -189,17 +189,31 @@ def generate_pods(self, job):
...
 
-    def generate_secret(self, p):
-        assert(isinstance(self.template, Template))
-        secret_yaml = self.secret_template.render(plugin=p)
+    def generate_secrets(self, job):
+        """generate_plugin_secrets must be called after generate_pods"""
+        assert (isinstance(job, Job))
+        params = job.params
+
+        if "plugin" not in params:
```
Comment (leigaoms) [inline src/ClusterManager/pod_template.py]: “plugins”

File: src/ClusterManager/job_launcher.py [hunk 3]
```diff
@@ -460,17 +526,25 @@ def submit_job(self, job):
...
+                pod_template = PodTemplate(job_object.get_template(),
+                                           deployment_template=job_object.get_deployment_template(),
+                                           enable_custom_scheduler=False,
+                                           secret_template=secret_template)
             else:
                 dataHandler.SetJobError(job_object.job_id, "ERROR: invalid jobtrainingtype: %s" % job_object.params["jobtrainingtype"])
                 dataHandler.Close()
                 return False
 
+            secrets = pod_template.generate_secrets()
```
Comment (leigaoms) [inline src/ClusterManager/job_launcher.py]: need to generate_pods() first? and not pass job to generate_secrets()

**Pull request merged**